src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java

```diff
@@ -605,20 +603,113 @@ public void disconnectCreatedInjection(InjectionCreationInfos modificationInfos,
         }
     }
 
-    public void modifyInjectionConnection(InjectionModificationInfos modificationInfos, Injection<?> injection) {
-        if (modificationInfos.getConnected() != null) {
-            if (injection.getTerminal().isConnected() && Boolean.FALSE.equals(modificationInfos.getConnected().getValue())) {
+    public ReportNode modifyInjectionConnectivityAttributes(ConnectablePosition<?> connectablePosition,
+                                                      ConnectablePositionAdder<?> connectablePositionAdder,
+                                                      Injection<?> injection,
+                                                      InjectionModificationInfos modificationInfos,
+                                                      ReportNode connectivityReports) {
+        List<ReportNode> reports = new ArrayList<>();
+        if (connectablePosition != null) {
+            modifyExistingConnectivityPosition(connectablePosition, modificationInfos, reports);
+        } else {
+            createNewConnectivityPosition(connectablePositionAdder, modificationInfos, reports);
+        }
+        modifyInjectionConnection(modificationInfos, injection, reports);
+        return reportModifications(connectivityReports, reports, "ConnectivityModified", CONNECTIVITY, Map.of());
+    }
+
+    private void modifyExistingConnectivityPosition(ConnectablePosition<?> connectablePosition,
+                                                  InjectionModificationInfos modificationInfos,
+                                                  List<ReportNode> reports) {
+        // Extract values with null checks
+        String connectionName = getValue(modificationInfos.getConnectionName());
+        ConnectablePosition.Direction connectionDirection = getValue(modificationInfos.getConnectionDirection());
+        Integer connectionPosition = getValue(modificationInfos.getConnectionPosition());
+
+        // Set properties and add reports
+        setFeederPropertiesAndReport(connectablePosition.getFeeder(), reports, connectionName, connectionDirection, connectionPosition);
+    }
+
+    private <T> T getValue(AttributeModification<T> attributeModification) {
+        return attributeModification != null ? attributeModification.getValue() : null;
+    }
+
+    private void setFeederPropertiesAndReport(ConnectablePosition.Feeder feeder,
+                                              List<ReportNode> reports,
+                                              String connectionName,
+                                              ConnectablePosition.Direction connectionDirection,
+                                              Integer connectionPosition) {
+        if (connectionName != null) {
+            feeder.setName(connectionName);
+            reports.add(buildModificationReport(null, connectionName, "Connection name"));
```

Comment:
If pass null as old value, the report can not show an old value, see suggestion in this PR
https://github.com/gridsuite/network-modification-server/pull/513